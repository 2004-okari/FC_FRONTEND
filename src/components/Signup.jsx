import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { signup } from '../redux/auth/authSlice';

const Signup = () => {
  const dispatch = useDispatch();
  const [username, setUsername] = useState('');

  const handleSignup = (e) => {
    e.preventDefault();
    dispatch(signup(username));
  };

  return (
    <form className="flex flex-col gap-8 items-center justify-center w-full h-screen" onSubmit={handleSignup}>
      <input
        type="text"
        name="username"
        autoComplete="username"
        placeholder="username"
        className="rounded-lg border-2 w-4/5 md:w-[50%] p-1"
        value={username}
        onChange={(e) => setUsername(e.target.value)}
      />

      <button
        type="submit"
        className="text-xl font-bold bg-orange-500 text-white rounded-lg py-1 w-24"
      >
        Sign up
      </button>
      <span className="font-bold">already user? Login</span>
    </form>
  );
};

export default Signup;
