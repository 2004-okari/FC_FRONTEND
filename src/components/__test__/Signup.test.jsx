import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import Signup from '../Signup';

const mockStore = configureStore([]);

describe('Signup Component', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      auth: {
        token: null,
      },
    });

    store.dispatch = jest.fn();

    component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <Signup />
        </MemoryRouter>
      </Provider>,
    );
  });

  it('renders correctly', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
