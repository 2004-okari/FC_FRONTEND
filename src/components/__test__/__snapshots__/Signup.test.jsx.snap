// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Component renders correctly 1`] = `
<form
  className="relative flex flex-col gap-8 items-center justify-center w-full h-screen"
  onSubmit={[Function]}
>
  <input
    autoComplete="username"
    className="rounded-lg border-2 w-4/5 md:w-[50%] p-1"
    name="username"
    onChange={[Function]}
    placeholder="Enter an username to sign up"
    type="text"
    value=""
  />
  <button
    className="text-xl bg-green-600 hover:bg-green-700 duration-300 text-white rounded-lg py-1 w-24"
    type="submit"
  >
    Sign up
  </button>
  <p>
    <span
      className="text-gray-400"
    >
      Already have an account? 
    </span>
    <a
      className="hover:underline font-bold"
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
  </p>
</form>
`;
